library/core/src/ptr/mod.rs

```diff
@@ -17,6 +17,9 @@
 //!
 //! * For memory accesses of [size zero][zst], *every* pointer is valid, including the [null]
 //!   pointer. The following points are only concerned with non-zero-sized accesses.
+//! * Volatile operations may act on memory outside Rust allocations, such as
+//!   I/O registers at hard-coded addresses. This case is exempt from the rules
+//!   below, but it does have its own rules documented at each volatile operation.
```

Comment:
Hm, I think I'd prefer to keep these docs here exclusive to actual Rust memory, and say everything volatile-specific in the volatile docs.

Reply:
I thought leaving the documentation as-was would open the door for ambiguity on whether these rules apply to volatile as well, so I added this point to specify it applies when acting on Rust memory, and otherwise has its own rules. I think we could remove this here, though, and say on the volatile docs that the module rules only apply to volatile in particular cases.

Reply:
The edits I pushed already do that, by saying that the ptr needs to be valid for reads *or* it can be entirely outside Rust memory.